src/generator/districts/test.rs

```diff
@@ -81,4 +82,215 @@ mod tests {
+            &mut rng,
+            &mut world,
+            &mut editor,
+            Some(0),
+            None, // No permit blocks
+            Some(false), // Ignore water
+        ).await;
+
+        editor.flush_buffer().await;
+    }
+
+    #[tokio::test]
+    async fn district_replace_ground_smooth() {
+        init_logger();
+
+        // Initialize the test data
+        let seed = Seed(12345);
+        let mut rng = RNG::new(seed);
+
+        let provider = GDMCHTTPProvider::new();
+
+        let build_area = provider.get_build_area().await.expect("Failed to get build area");
+
+        let mut editor = Editor::new(build_area);
+        let mut world = World::new(&provider).await.expect("Failed to create world");
+
+        let _districts = generate_districts(seed, &mut world).await;
+
+        let block_vec = vec![
+            Block {
+                id: BlockID::Stone,
+                data: None,
+                state: None,
+            },
+            Block {
+                id: BlockID::Cobblestone,
+                data: None,
+                state: None,
+            },
+            Block {
+                id: BlockID::StoneBricks,
+                data: None,
+                state: None,
+            },
+            Block {
+                id: BlockID::Andesite,
+                data: None,
+                state: None,
+            },
+            Block {
+                id: BlockID::Gravel,
+                data: None,
+                state: None,
+            },
+            Block {
+                id: BlockID::StoneStairs,
+                data: None,
+                state: None,
+            },
+            Block {
+                id: BlockID::CobblestoneStairs,
+                data: None,
+                state: None,
+            },
+            Block {
+                id: BlockID::StoneBrickStairs,
+                data: None,
+                state: None,
+            },
+            Block {
+                id: BlockID::AndesiteStairs,
+                data: None,
+                state: None,
+            },
+            Block {
+                id: BlockID::StoneSlab,
+                data: None,
+                state: None,
+            },
+            Block {
+                id: BlockID::CobblestoneSlab,
+                data: None,
+                state: None,
+            },
+            Block {
+                id: BlockID::StoneBrickSlab,
+                data: None,
+                state: None,
+            },
+            Block {
+                id: BlockID::AndesiteSlab,
+                data: None,
+                state: None,
+            },
+        ];
+
+        let mut blocks_dict: HashMap<u32, HashMap<u32, f32>> = HashMap::new();
+
+        let block_dict = [
+            (0, 3.0),  // Stone
+            (1, 2.0),  // Cobblestone
+            (2, 8.0),  // Stone Bricks
+            (3, 3.0),  // Andesite
+            (4, 1.0),  // Gravel
+        ].iter().cloned().collect();
```

Comment:
instead of iter().cloned() you can write into_iter()

iter() creates an iterator of references, which must be cloned if you want the original values

into_iter() actually consumes the original container and gives you an iterator of the original values, which is what you want